src/Fusion.Summary.Api/Domain/Queries/GetDepartment.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Fusion.Summary.Api.Database;
+using Fusion.Summary.Api.Database.Models;
+using Fusion.Summary.Api.Domain.Models;
+using Fusion.Summary.Api.Services;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Fusion.Summary.Api.Domain.Queries;
+
+public class GetDepartment : IRequest<QueryDepartment?>
+{
+    public string SapDepartmentId { get; set; }
+
+    public GetDepartment(string sapDepartmentId)
+    {
+        SapDepartmentId = sapDepartmentId;
+    }
+
+    public class Handler : IRequestHandler<GetDepartment, QueryDepartment?>
+    {
+        private readonly DatabaseContext _context;
+
+        public Handler(DatabaseContext context)
+        {
+            _context = context;
+        }
+
+        public async Task<QueryDepartment?> Handle(GetDepartment request, CancellationToken cancellationToken)
+        {
+            // Get all departments 
+            var dbDepartments = await _context.Departments.ToListAsync();
```

Comment:
Could we use FirstOrDefaultAsync here to avoid retrieving all deps

Reply:
Fixed